src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHostBuilder.cs

```diff
@@ -14,23 +14,27 @@ namespace Microsoft.AspNetCore.Hosting
     public interface IWebHostBuilder
     {
         /// <summary>
-        /// Builds an <see cref="IWebHost"/> which hosts a web application.
+        /// The <see cref="WebHostBuilderContext"/> used during building.
         /// </summary>
-        IWebHost Build();
+        /// <remarks>
+        /// Some properties of this type will be null whilst it is being built, most noteably the <see cref="ILoggerFactory"/> will
+        /// be null inside the <see cref="ConfigureAppConfiguration(Action{WebHostBuilderContext, IConfigurationBuilder})"/> method.
+        /// </remarks>
+        WebHostBuilderContext Context { get; }
```

Comment:
@Tratcher @glennc @davidfowl @DamianEdwards 

I feel that there is little benefit of having this context on the WebHostBuilder itself since its properties won't be initialized until Build(). We are providing the context in the other methods and extensions now and the WebHostBuilderContext will be resolvable from DI where needed. I plan to remove this, but just wanted to make sure everyone is on the same page first.